app/models/user_role.rb

```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class UserRole < ApplicationRecord
+  belongs_to :user
```

Comment:
You could consider using Polymorphic associations here.  Although I think all the models could do with a thorough review -if we had more than a few days left!

Reply:
This is a join table rather than an either-or relationship. That's a new thing to me in rails though, nice feature.  Worth suggesting for the 'needs can be assigned to a user or a role' requirement!